waspc/cli/src/Wasp/Cli/Command/CreateNewProject/StarterTemplates/StarterTemplate.hs

```diff
@@ -0,0 +1,61 @@
+module Wasp.Cli.Command.CreateNewProject.StarterTemplates.StarterTemplate
+  ( StarterTemplate (..),
+    TemplateMetadata (..),
+    StartingInstructionsBuilder,
+    WaspAppAiGenerator (..),
+    getTemplateStartingInstructions,
+    getTemplateName,
+  )
+where
+
+import StrongPath (Dir', Path', Rel')
+import qualified Wasp.Cli.GithubRepo as GhRepo
+import qualified Wasp.Cli.Interactive as Interactive
+
+data StarterTemplate
+  = -- | Template from a Github repo.
+    GhRepoStarterTemplate !GhRepo.GithubRepoRef !TemplateDirPath !TemplateMetadata
+  | -- | Template from a disk, that comes bundled with wasp CLI.
+    LocalStarterTemplate !TemplateDirPath !TemplateMetadata
+  | -- | Template that will be dynamically generated by AI based on user's input.
+    AiGeneratedStarterTemplate !WaspAppAiGenerator !TemplateMetadata
+
+-- | Path to a directory containing template files.
+type TemplateDirPath = Path' Rel' Dir'
+
+data TemplateMetadata = TemplateMetadata
+  { _tmplName :: !String,
+    _tmplDescription :: !String,
+    _tmplBuildStartingInstructions :: !StartingInstructionsBuilder
+  }
+
+-- | AI generators that can generate Wasp apps.
+data WaspAppAiGenerator = WaspAI
```

Comment:
Are we overengineerring here a bit? Are we working on other AI generators for Wasp or looking to support other AI external AI generators? Sorry if I missed that we are doing something like that.

Reply:
Yes, that is my secret project :D! Kidding, it is a tiny bit of overengienerg that allowed me to treat AI starter template same as the rest while keeping compile time safety. Commented on it also in another comment in more details.